src/garbage_collector.rs

```diff
@@ -20,6 +25,34 @@ enum GarbageCollectorMessage {
     Shutdown,
 }
 
+pub async fn run_garbage_collector_once(
```

Comment:
can we not have this specify that the gc should run once? It can just be a fn we expose for running the gc in the foreground, as opposed to the background (which the `GarbageCollector` struct does). Also can we have the caller pass in the manifest/tablestore/stats to keep it consistent with `GarbageCollector`?